src/nile/core/declare.py

```diff
@@ -15,8 +15,10 @@ def declare(contract_name, network, alias=None, overriding_path=None):
 
     output = run_command(contract_name, network, overriding_path, operation="declare")
     class_hash, tx_hash = parse_information(output)
-    logging.info(f"⏳ Declaration of {contract_name} successfully sent at {class_hash}")
-    logging.info(f"🧾 Transaction hash: {tx_hash}")
+    logging.info(
+        f"⏳ Declaration of {contract_name} successfully sent at {hex(class_hash)}"
+    )
+    logging.info(f"🧾 Transaction hash: {hex(tx_hash)}")
 
     deployments.register_class_hash(class_hash, network, alias)
```

Comment:
why isn't the class hash stored as hex too? it should be considered external to me, since it's available to users/plugins/external tooling

Reply:
The class hash is stored as hex (is being converted in the `register_class_hash` following the logging lines).

It is not converted here and passed as an argument because that is against the pattern (the current definition, at least), which states:

- Accepting and using only integers for Internal APIs (register_class_hash is an internal API)
- Convert to hex before writing to files (we are doing that inside the method)

This is the current definition of the pattern: https://github.com/OpenZeppelin/nile/issues/216